python/lsst/pipe/base/graph/graph.py

```diff
@@ -173,6 +170,15 @@ def _buildGraphs(
             # a newly created QuantumNode to the appropriate input/output
             # field.
             for quantum in quantumSet:
+                if quantum.dataId is not None:
```

Comment:
I'm not really bothered by codecov's complaints here: I'm just being super defensive about trying to avoid backwards-compatibilty breakage by guessing the `DimensionUniverse`, but really everybody should be using the code path that is tested, in which the universe is passed explicitly (since that's what `GraphBuilder`).